utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -705,33 +939,53 @@ pub struct CodePointMapRangeIterator<'a, T: TrieValue> {
     // When `range` == `None`, it means that we have hit the end of iteration. It would occur
     // after a call to `next()` returns a None <=> we attempted to call `get_range()`
     // with a start code point that is > CODE_POINT_MAX.
-    range: Option<CodePointMapRange>,
+    cpm_range: Option<CodePointMapRange<T>>,
 }
 
 impl<'a, T: TrieValue + Into<u32>> Iterator for CodePointMapRangeIterator<'a, T> {
-    type Item = CodePointMapRange;
+    type Item = CodePointMapRange<T>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        match self.range {
-            Some(CodePointMapRange { start, end, .. }) => {
-                let next_range = if start == u32::MAX {
+        // In order to work around Rust disallowance of Copy impls for Range in Rust,
```

Comment:
This is clunky.  I pushed a commit to your branch to streamline it.

Reply:
Thanks. I didn't realize that calling `.clone()` was an option here -- I assumed it was "bad" (cause duplication of values on the stack from or "worse") and to be avoided at all costs.